blocks/dealer-locator/sidebar-maps.js

```diff
@@ -528,133 +541,96 @@ $.fn.isWaypoint = function (waypoint) {
-      end.setMinutes(moment(endTime, ["h:mm A"]).format("mm"));
-      end.setSeconds(0);
-
-      if (~endTime.toLowerCase().indexOf("am")) {
-        console.log('end date is AM, add 1 day');
-        end.setDate(time.getDate() + 1);
+    if (day.toLowerCase() === 'closed') {
+      dayObj = {
+        Start: 'Closed',
+        End: 'Closed',
       }
+    } else {
 
-      if (moment.tz(zone).isBetween(start, end)) {
-
-        hours = Math.abs(time - end) / 36e5;
+      var hours = day.split('-');
+      hours.forEach((time, idx) => {
+        var hour = parseInt(time.substr(0,2));
+        var minutes = time.substr(2);
+        var meridian = 'AM';
 
-        if (hours < 1) {
-          closeSoon = true;
+        if (hour > 12) {
+          hour = hour - 12;
+          meridian = 'PM';
         }
 
-        return { open: true, endTime: end, closeSoon: closeSoon };
-
-      } else {
-        return { open: false, endTime: end, closeSoon: closeSoon };
-      }
+        if (idx === 0) {
+          dayObj.Start = `${hour}:${minutes} ${meridian}`;
+        } else if (idx === 1) {
+          dayObj.End = `${hour}:${minutes} ${meridian}`;
+        };
+      });
     }
-
-
-  }
-
-  return 2;
-
+    week.push(dayObj);
+  });
+  return week;
 };
 
-$.fn.canDetermineHours = function (pin) {
+$.fn.getOpenHours = function (pin) {
+  var time = new Date();
+  var today = time.getDay();
 
-  var hours = null;
+  var earliestHour;
+  var latestHour;
 
-  if (pin.hours['Parts']) {
-    hours = pin.hours['Parts'];
-  }
+  if (pin.hours.length === 0) {
+    earliestHour = '';
+    latestHour = '';
+  } else {
+    var allWeek = $.fn.buildWeek(pin.hours);
+    var openHours = allWeek[today]
 
-  if (!hours && pin.hours['Service']) {
-    hours = pin.hours['Service'];
+    earliestHour = openHours.Start;
+    latestHour = openHours.End;
   }
+  return { open: earliestHour, close: latestHour }
+};
 
-  if (!hours && pin.hours['Leasing']) {
-    hours = pin.hours['Leasing'];
+$.fn.isOpen = async function (dealer, time) {
+  var hours = $.fn.getOpenHours(dealer);
+  var compareDate = '1/1/2000 '
+  
+  if (!dealer.timeZoneId) {
+    dealer.timeZoneId = await $.fn.getTimeZoneId(dealer);
   }
 
   if (hours) {
+    var start = new Date(compareDate + hours.open);
+    var end = new Date(compareDate + hours.close);
 
-    return true;
+    if (~hours.close.toLowerCase().indexOf("am")) {
+      end.setDate(compareDate.getDate() + 1);
+    }
+
+    var openTime = (start.getHours() * 60) + start.getMinutes();
+    var closeTime = (end.getHours() * 60) + end.getMinutes();
+
+    var stringDealerDate = moment().tz(dealer.timeZoneId).format();
+    var hourPosition = stringDealerDate.indexOf('T');
+    var dealerLocalHour = stringDealerDate.substring(hourPosition + 1, hourPosition + 6);
+    var [ hour, minutes ] = dealerLocalHour.split(':');
+    var dealerTime = (Number(hour) * 60) + Number(minutes);
+
+    if (dealerTime >= openTime && dealerTime < closeTime) {
```

Comment:
this is ok as it is but this "if" also can be coded as the following:
return { close: (dealerTime >= openTime && dealerTime < closeTime) }
WDYT? 🤔